### ISSUE TITLE
tensorboard tensor visualisation -- hooks

### ISSUE THREAD

**Issue by JohannesMaxWel** [uclmr/jack#182]
Hi,

I was just looking into using tensorboard to visualise the ongoings of the tensorflow models.

Some of our Hooks have a `summary_writer` attribute which can be used to dump the relevant information into a log file. So far the `summary_writer` works great for scalar values like training speed or the evaluation metrics. I want to use this `summary_writer` for visualising tensors.

Functionalities in the hooks to log the evolution of embeddings, matrices and other tensors during training would be great, and tensorboard has some pretty awesome visualisations.

But currently all hooks are called during training with the same _loss_ input argument, and don't generally allow for other tensors or summaries which the hooks could log `at_iteration_end` or `at_epoch_end`, even if they were passed over from the `train()` function.
In the way the hooks are currently set up they only allow for logging scalars related to the loss, not arbitrary summary writer operations.

How could we change this? I only see a redefinition of the Hooks abstraction, but perhaps someone has already done some work here..?

Cheers 
Johannes

**Comment by dirkweissenborn** [comment]
summaries can be written with tf.summary, they are fetched every iteration and can be inspected in the tensorboard.
